hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2785,3 +2794,224 @@ func reconcileMachineConfigServerService(svc *corev1.Service) error {
 	svc.Spec.ClusterIP = corev1.ClusterIPNone
 	return nil
 }
+
+func (r *HostedClusterReconciler) reconcileMachineApprover(ctx context.Context, hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) error {
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(controlPlaneNamespace), controlPlaneNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane namespace: %w", err)
+	}
+
+	// Reconcile machine-approver role
+	role := machineapprover.Role(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
```

Comment:
can we update all this calls to createOrUpdate to match this pattern?
```
	tokenSecret := TokenSecret(controlPlaneNamespace, nodePool.Name, targetConfigVersionHash)
	if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, tokenSecret, func() error {
		return reconcileTokenSecret(tokenSecret, nodePool, compressedConfig)
	}); err != nil {
		return ctrl.Result{}, fmt.Errorf("failed to reconcile token Secret: %w", err)
	} else {
		log.Info("Reconciled token Secret", "result", result)
		span.AddEvent("reconciled token Secret", trace.WithAttributes(attribute.String("result", string(result))))
	}
```

Reply:
so inline err handling is preferred? sure, will update 👍